File: develop/binder-canary/src/main/java/com/zipper/develop/binder/canary/BinderCanary.java
package com.zipper.develop.binder.canary;

import androidx.annotation.NonNull;

/**
 * @author zhangzhipeng
 * @date 2023/7/7
 */
public class BinderCanary {

    static {
        System.loadLibrary("canary");
    }

    public static native boolean init(@NonNull MonitorConfig monitorConfig);

}

File: develop/binder-canary/src/main/java/com/zipper/develop/binder/canary/BinderCanary.kt
package com.zipper.develop.binder.canary

/**
 * @author zhangzhipeng
 * @date 2023/7/7
 */
object BinderCanary {
    init {
        System.loadLibrary("canary")
    }

    @JvmStatic
    external fun init(monitorConfig: MonitorConfig): Boolean

    @JvmStatic
    external fun monitor(): Boolean

    @JvmStatic
    external fun unmonitored(): Boolean

    @JvmStatic
    fun onReport(errorCode: Int, dataSize: Int, backtrace: String) {

    }
}
